src/coreclr/nativeaot/Runtime/CachedInterfaceDispatchAot.cpp

```diff
@@ -0,0 +1,54 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "common.h"
+#include "CachedInterfaceDispatchPal.h"
+#include "CachedInterfaceDispatch.h"
+
+// The base memory allocator.
+static AllocHeap * g_pAllocHeap = NULL;
+
+bool InterfaceDispatch_InitializePal()
+{
+    g_pAllocHeap = new (nothrow) AllocHeap();
+    if (g_pAllocHeap == NULL)
+        return false;
+
+    if (!g_pAllocHeap->Init())
+        return false;
+
+    return true;
+}
+
+// Allocate memory aligned at sizeof(void*)*2 boundaries
+void *InterfaceDispatch_AllocDoublePointerAligned(size_t size)
+{
+    return g_pAllocHeap->AllocAligned(size, sizeof(void*) * 2);
+}
+
+// Allocate memory aligned at at least sizeof(void*)
+void *InterfaceDispatch_AllocPointerAligned(size_t size)
+{
+    return g_pAllocHeap->Alloc(size);
```

Comment:
```suggestion
    return g_pAllocHeap->AllocAligned(size, sizeof(void*));
```

I get that perhaps the `Alloc()` provides this guarantee, but it looks odd for one to call out alignment and use an explicit API and the other providing a similar guarantee and calling a seemingly more general function. It also looks like `Alloc()` calls `AllocAligned()` and passes `1`, so now I have more questions.

Reply:
This is a copy paste/from the old logic. Of course, now that I'm looking at this, I'm pretty convinced that the Alloc api on AllocHeap is basically incorrect to use in most cases, but we get away with it since it is ONLY called with objects of type `DiscardedCacheBlock`  (which ironically, is a pair of pointers), and since we start pointer aligned, we're always still pointer aligned in the `AllocHeap`. I'll convert it to `AllocAligned`, but it won't actually change any meaningful behavior anywhere. Thankfully.